models/branches.go

```diff
@@ -26,11 +26,11 @@ const (
 
 // ProtectedBranch struct
 type ProtectedBranch struct {
-	ID                        int64  `xorm:"pk autoincr"`
-	RepoID                    int64  `xorm:"UNIQUE(s)"`
-	BranchName                string `xorm:"UNIQUE(s)"`
-	CanPush                   bool   `xorm:"NOT NULL DEFAULT false"`
-	EnableWhitelist           bool
+	ID                        int64              `xorm:"pk autoincr"`
+	RepoID                    int64              `xorm:"UNIQUE(s)"`
+	BranchName                string             `xorm:"UNIQUE(s)"`
+	CanPush                   bool               `xorm:"NOT NULL DEFAULT false"`
+	EnableWhitelist           bool               `xorm:"NOT NULL DEFAULT false"`
```

Comment:
Bools always default to false, but why this column needs NOT NULL now? The `Sync2()` function will not update the column in a migration, AFAIK (didn't in my database).

Reply:
Not null is needed so not to have null values in database

Reply:
> 
> 
> Not null is needed so not to have null values in database

@lafriks I know, but this column was always nullable and worked fine. If we suddenly need to make it not null, we need to create some kind of special migration (`ALTER TABLE`) because xorm's `Sync2()` won't touch the column if it already exists. I've tested that locally.

Reply:
I just found it strange only this variable didn't have any xorm directive (NOT NULL), I don' think it will make any difference? Should I remove it again or try to add something to the migration?

Reply:
> 
> 
> I just found it strange only this variable didn't have any xorm directive (NOT NULL), I don' think it will make any difference? Should I remove it again or try to add something to the migration?

In all formality, it should be `NOT NULL` (which requires `DEFAULT false`). But it was working so far, so.... The thing is I'm not sure what xorm does behind the curtains (it needs to be checked for each engine). In plain SQL, the following needs to be considered:

* Some databases support bool values; some don't, so tinyint or similar is used instead.
* Some databases resolve `WHERE null = false` as true, some don't.
* Some databases resolve `WHERE null <> true` as true, some don't.

So the question goes beyond this PR. I think that problems may show up if the column is used for filtering (i.e. in a `WHERE` clause). Otherwise it will be no problem. I just wanted to make you aware that this change may not give the expected results.

Reply:
I think ALTER TABLE will not be possible for sqlite (https://stackoverflow.com/questions/29583396/modify-sqlite-table-column-not-null-to-null)

Thus it would require to create a new column and replace with the old one. Currently I don't think this is an issue, but if new features are implemented where this is not considered it can result in bugs. So is it better to leave it without the NOT NULL to not mislead?

Reply:
> 
]> 
> So is it better to leave it without the NOT NULL to not mislead?

I think that's preferrable, yes.